### PR TITLE
Rework _get_axnpby_kerns to avoid the use of matrix banks.

### PR DESCRIPTION
**Pull request by FreddieWitherden**

### CODE REVIEW

File: pyfr/plugins/integrate.py [hunk 1]
```diff
@@ -107,26 +107,9 @@ def _init_gradients(self, intg, rinfo):
...
-                # Get the smats at the solution points and subset
-                smat = eles.smat_at_np('upts')[..., eset]
-
-                # Get |J|^-1 at the solution points and subset
-                rcpdjac = eles.rcpdjac_at_np('upts')[:, eset]
-
-                # Product to give J^-T at the solution points
-                self._rcpjact.append(rcpdjac*smat.transpose(2, 0, 1, 3))
-
     def _eval_exprs(self, intg):
```
Comment (WillTrojak) [inline pyfr/plugins/integrate.py:113]: Should this change to the gradient handling be in a different PR?

File: pyfr/plugins/tavg.py [hunk 2]
```diff
@@ -74,27 +74,9 @@ def _init_gradients(self, intg):
             gradpnames.update(re.findall(r'\bgrad_(.+?)_[xyz]\b', ex))
 
         privarmap = self.elementscls.privarmap[self.ndims]
-        self._gradpinfo = [(pname, privarmap.index(pname)) 
+        self._gradpinfo = [(pname, privarmap.index(pname))
                             for pname in gradpnames]
```
Comment (WillTrojak) [inline pyfr/plugins/tavg.py:79]: Should this change to the gradient handling be in a different PR?

**Review by WillTrojak — COMMENTED** [review]
Other than moving the plugin changes to a different PR, this looks fine.

File: pyfr/plugins/integrate.py (hunk 1 above)
Comment (FreddieWitherden) [reply pyfr/plugins/integrate.py:113]: It isn't a change per-se, just removing some dead code (the arrays are assigned to, but never read).

**Comment by semi-h** [comment]
https://github.com/PyFR/PyFR/blob/9634d2aa7798262a0aa0420e2e03be437bea5857/pyfr/integrators/std/controllers.py#L125
https://github.com/PyFR/PyFR/blob/9634d2aa7798262a0aa0420e2e03be437bea5857/pyfr/integrators/dual/pseudo/pseudocontrollers.py#L28

Why didn't you get rid of these prepare_reg_banks calls as well?

**Comment by FreddieWitherden** [comment]
> https://github.com/PyFR/PyFR/blob/9634d2aa7798262a0aa0420e2e03be437bea5857/pyfr/integrators/std/controllers.py#L125
> 
> https://github.com/PyFR/PyFR/blob/9634d2aa7798262a0aa0420e2e03be437bea5857/pyfr/integrators/dual/pseudo/pseudocontrollers.py#L28
> 
> Why didn't you get rid of these prepare_reg_banks calls as well?

These are for the errest kernels, rather than axnpby.  Their time will come, however.

**Review by semi-h — APPROVED** [review]
